### PR TITLE
Shorten git hash in reproducibility table

### PR DESCRIPTION
**Pull request by slager**
This shortens the git hash in the reproducibility table from 40 to 7.  7 is the standard abbreviated hash length used by GitHub and by the git command line tools (e.g., `git log --oneline`).

It's not clear to me why the line above my change is replacing `(@` with `(`, but that pattern does not seem to be matched in my testing, so it looks like it is not currently doing anything.  Maybe one of you reviewers will know its intent and whether we should leave it or remove it?

@kelliemac, I'm just seeing now that you also have a branch here that shortens the hash, so this might be partially duplicating that?

### CODE REVIEW

**Comment by kelliemac** [comment]
thanks for tackling this @slager! would you mind adding a comment to the code that explains the regular expression you're using in plain english? I always find these things hard to parse and I think it will probably help us later on to have a description there.

**Comment by kelliemac** [comment]
@mayerbry do you know why the "@" is replaced in the table? I am guessing it is either due to pdf/word rendering issues or because the "#" was intended to make it clear that the string following it is a hash, but I'm not sure

**Comment by kelliemac** [comment]
and yes @slager this overlaps with my branch but I think your approach is cleaner so I'm happy to go with this and delete my branch once we merge this PR!

**Comment by slager** [comment]
To be clear, I don't think the `@` was actually getting replaced in the table by the code above my changes, because that would only happen if the `@` came immediately after a `(`.  That does not seem to ever be the case, at least when `source` involves a git hash.

**Comment by kelliemac** [comment]
> To be clear, I don't think the `@` was actually getting replaced in the table by the code above my changes, because that would only happen if the `@` came immediately after a `(`. That does not seem to ever be the case, at least when `source` involves a git hash.

Yes I agree. I think in my branch there is a fix that actually replaces the "@" if we want to do that. Otherwise, I am in favor of just deleting that line.

**Review by kelliemac — APPROVED** [review]

**Comment by slager** [comment]
Yeah, and fwiw I kind of like keeping the `@` before the shortened hash as a way to separate the hash from the repo URL.  We can see what Bryan says. If we don't need it I can push another commit that gets rid of the line before my changes.

**Comment by kelliemac** [comment]
@slager anything outstanding for this PR before it can be merged?

**Comment by slager** [comment]
Yes @kelliemac , before merging hoping to hear @mayerbry 's opinion on whether we're good to delete this part, which I don't think does anything:

https://github.com/FredHutch/VISCfunctions/blob/c48a95ccebcf277c15bca718bd0d8d750bb8d6c0/R/reproducibility_tables.R#L197-L198

**Comment by mayerbry** [comment]
I was looking at my reports and it looks like there is always a `@` before the git hash. Is this not true of other reports?

It does seem that one line of gsub code doesn't do anything, so fine to remove

**Comment by slager** [comment]
Yes, there is always a `@` before the hash, but there's never a `(` before the `@`, so I don't think this line/regex makes sense:

https://github.com/FredHutch/VISCfunctions/blob/c48a95ccebcf277c15bca718bd0d8d750bb8d6c0/R/reproducibility_tables.R#L197-L198

I'll push a commit shortly to remove this line.

**Review by mayerbry — APPROVED** [review]
